http-client-spi/src/main/java/software/amazon/awssdk/http/SdkHttpConfigurationOption.java

```diff
@@ -131,6 +131,16 @@
     public static final SdkHttpConfigurationOption<Duration> TLS_NEGOTIATION_TIMEOUT =
         new SdkHttpConfigurationOption<>("TlsNegotiationTimeout", Duration.class);
 
+    /**
+     * Configure whether to use a non-blocking dns resolver or not. False by default, as netty's default dns resolver is
+     * blocking; it namely calls java.net.InetAddress.getByName.
+     * <p>
+     * When enabled, a non-blocking dns resolver will be used instead, by modifying netty's bootstrap configuration.
+     * See https://netty.io/news/2016/05/26/4-1-0-Final.html
+     */
+    public static final SdkHttpConfigurationOption<Boolean> USE_NONBLOCKING_DNS_RESOLVER =
+        new SdkHttpConfigurationOption<>("UseNonBlockingDnsResolver", Boolean.class);
```

Comment:
Let's leave this off of the default options, since no other HTTP clients use it yet. Prior art: https://github.com/aws/aws-sdk-java-v2/blob/7d302ee48653482b71ab3e650c1b2edb6aa0b4c6/http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/NettyNioAsyncHttpClient.java#L662-L666

Reply:
Done.